CMakeLists.txt

```diff
@@ -337,6 +337,26 @@ endif (WIN32)
 
 endif (OVERRIDE_CMAKE_CXX_FLAGS)
 
+OPTION(CMAKE_ENABLE_CODE_COVERAGE "Enable code coverage" OFF)
+
+if(CMAKE_ENABLE_CODE_COVERAGE)
+
+  string(TOUPPER ${CMAKE_BUILD_TYPE} UPPERCASE_CMAKE_BUILD_TYPE)
+  if(NOT UPPERCASE_CMAKE_BUILD_TYPE STREQUAL DEBUG)
```

Comment:
The stuff we already have in differentiates between windows and Unix, considering windows multi-target and Unix single-target. So it behaves correctly in both cases.
We should actually modify that at some point to make it work with multi-target generators on Unix as well, but it is a low prio at this moment.